### ISSUE TITLE
UrlBuilder queryParameters

### ISSUE THREAD

**Issue by rickle-msft** [Azure/autorest-clientruntime-for-java#359]
The addQueryParameter method should overwrite existing parameters with the same key instead of duplicating them (which the method looks to do now since it will just append the supplied key and value). Possibly rename the method to setQueryParameter to make this behavior clear.

Side note: Should the UrlBuilder class be final?

**Comment by daschult** [comment]
Yes, I guess UrlBuilder should be final. I usually don't think about that, but it does make sense here.
